src/mpid/ch4/netmod/ofi/ofi_win.c

```diff
@@ -144,15 +144,19 @@ static int win_allgather(MPIR_Win * win, void *base, int disp_unit)
     int rc = 0, allrc = 0;
     MPIDI_OFI_WIN(win).mr = NULL;
     if (base || (win->create_flavor == MPI_WIN_FLAVOR_DYNAMIC && !MPIDI_OFI_ENABLE_MR_ALLOCATED)) {
-        MPIDI_OFI_CALL_RETURN(fi_mr_reg(MPIDI_OFI_global.ctx[0].domain, /* In:  Domain Object */
-                                        base,   /* In:  Lower memory address */
-                                        win->size,      /* In:  Length              */
-                                        FI_REMOTE_READ | FI_REMOTE_WRITE,       /* In:  Expose MR for read  */
-                                        0ULL,   /* In:  offset(not used)    */
-                                        MPIDI_OFI_WIN(win).mr_key,      /* In:  requested key       */
-                                        0ULL,   /* In:  flags               */
-                                        &MPIDI_OFI_WIN(win).mr, /* Out: memregion object    */
-                                        NULL), rc);     /* In:  context             */
+        /* device buffers are not currently supported */
+        if (MPIR_GPU_query_pointer_is_dev(base))
+            rc = 1;
+        else
+            MPIDI_OFI_CALL_RETURN(fi_mr_reg(MPIDI_OFI_global.ctx[0].domain,     /* In:  Domain Object */
+                                            base,       /* In:  Lower memory address */
+                                            win->size,  /* In:  Length              */
+                                            FI_REMOTE_READ | FI_REMOTE_WRITE,   /* In:  Expose MR for read  */
+                                            0ULL,       /* In:  offset(not used)    */
+                                            MPIDI_OFI_WIN(win).mr_key,  /* In:  requested key       */
+                                            0ULL,       /* In:  flags               */
+                                            &MPIDI_OFI_WIN(win).mr,     /* Out: memregion object    */
+                                            NULL), rc); /* In:  context             */
     } else if (win->create_flavor == MPI_WIN_FLAVOR_DYNAMIC) {
```

Comment:
We will need do the same for dynamic windows at `win_attach_hook`.

Reply:
will add